src/fhir/api.test.ts

```diff
@@ -3,12 +3,160 @@ import { createBundle } from "./api";
 import { initialValues, initialWithNoVariant } from "../components/reports/FormDefaults";
 import { Observation } from "fhir/r4";
 import { geneCoding } from "../code_systems/hgnc";
+import { Bundle } from "@smile-cdr/fhirts/dist/FHIR-R4/classes/bundle";
+import { BundleEntry } from "@smile-cdr/fhirts/dist/FHIR-R4/classes/bundleEntry";
+import { Patient } from "@smile-cdr/fhirts/dist/FHIR-R4/classes/models-r4";
 
 const fhir = new Fhir();
 
 const reportedGenes = [geneCoding("HGNC:4389", "GNA01")];
+const FHIR_URL = process.env.REACT_APP_FHIR_URL || "";
+
+const checkResponseOK = async (response: Response) => {
+  const r = await response.json();
+
+  if (!response.ok) {
+    console.error(r.body);
+    throw new Error(response.statusText);
+  }
+  if (!(r.type === "Response")) {
+    return r;
+  }
+  type BundleResponse = {
+    entry: [
+      {
+        response: {
+          status: string | number;
+          outcome: {
+            issue?: [
+              {
+                diagnostics: string;
+              },
+            ];
+          };
+        };
+      },
+    ];
+  };
+  const bundleResponse = r as BundleResponse;
+
+  const errors = bundleResponse.entry
```

Comment:
ooh we could even do this I think

```suggestion
  const errors = (r as BundleResponse).entry
```